src/sync.cpp

```diff
@@ -511,10 +511,127 @@ bool assignFilesystemIds(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, h
     return success;
 }
 
+SyncConfigBag::SyncConfigBag(DbAccess& dbaccess, FileSystemAccess& fsaccess, PrnGen& rng)
+{
+    std::string dbname = "syncconfig";
+    mTable.reset(dbaccess.open(rng, &fsaccess, &dbname, false, false));
+    if (!mTable)
+    {
+        assert(false);
+        LOG_err << "Unable to open DB table: " << dbname;
+        return;
+    }
+
+    mTable->rewind();
+
+    decltype(mNextTableId) tableId;
+    std::string data;
+    while (mTable->next(&tableId, &data))
+    {
+        auto syncConfig = SyncConfig::unserialize(data);
+        if (!syncConfig)
+        {
+            assert(false);
+            LOG_err << "Unable to unserialize sync config at id: " << tableId;
+            continue;
+        }
+
+        mSyncConfigs.insert(std::make_pair(syncConfig->getLocalPath(), SyncConfigData{tableId, *syncConfig}));
+        if (tableId > mNextTableId)
+        {
+            mNextTableId = tableId;
+        }
+    }
+    ++mNextTableId;
+}
+
+void SyncConfigBag::add(const SyncConfig& syncConfig)
+{
+    auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
+    if (syncConfigPair == mSyncConfigs.end())
+    {
+        mSyncConfigs.insert(std::make_pair(syncConfig.getLocalPath(), SyncConfigData{mNextTableId, syncConfig}));
+        if (mTable)
+        {
+            auto data = syncConfig.serialize();
+            if (!mTable->put(mNextTableId, &data))
+            {
+                assert(false);
+                LOG_err << "Incomplete database put at id: " << mNextTableId;
+            }
+            ++mNextTableId;
+        }
+    }
+    else
+    {
+        update(syncConfig);
+    }
+}
+
+void SyncConfigBag::remove(const SyncConfig& syncConfig)
+{
+    auto syncConfigPair = mSyncConfigs.find(syncConfig.getLocalPath());
+    if (syncConfigPair != mSyncConfigs.end())
+    {
+        if (mTable)
+        {
+            if (!mTable->del(syncConfigPair->second.mTableId))
+            {
+                assert(false);
+                LOG_err << "Incomplete database del at id: " << syncConfigPair->second.mTableId;
+            }
+        }
+        mSyncConfigs.erase(syncConfig.getLocalPath());
+    }
+    else
+    {
+        assert(false && "sync config does not exist");
```

Comment:
Perhaps log an error message is also a good idea for debugging purposes (based on logs, rather on a debugger :) )

Reply:
Wouldn't it be better to adjust the logout procedure to not reach this point during logout?? Same for `update()`. If you added an `assert` in that place it is because you do expect to find the sync-config when the function is called. If it does not exist, it would represent a error. 
In conclusion, I suggest whether the logout is changed to avoid this failed assertion, or at least to log an error message, so a developer can notice the sync-config was not found.

Reply:
What I meant is that it's now a possible and expected program flow to erase a sync config that doesn't exist anymore. When you logout the sync config bag is cleared first and then all sync objects are destroyed which in turn attempts to remove their config from the bag (but it's already gone at this point).